doc/source/operations/gpu-in-openstack.rst

```diff
@@ -0,0 +1,836 @@
+        handlers:
+          - name: reboot
+            set_fact:
+              kayobe_needs_reboot: true
+
+      - name: Reboot when required
+        hosts: iommu:vgpu
+        tags:
+          - reboot
+        tasks:
+          - name: Reboot
+            reboot:
+              reboot_timeout: 3600
+            become: true
+            when: kayobe_needs_reboot | default(false) | bool
+
+Ansible Inventory Configuration
+-------------------------------
+
+Add some hosts into the ``vgpu`` group. The example below maps two custom
+compute groups, ``compute_multi_instance_gpu`` and ``compute_vgpu``,
+into the ``vgpu`` group:
+
+.. code-block:: yaml
+   :caption: $KAYOBE_CONFIG_PATH/inventory/custom
+
+    [compute]
+    [compute_multi_instance_gpu]
+    [compute_vgpu]
+
+    [vgpu:children]
+    compute_multi_instance_gpu
+    compute_vgpu
+
+    [iommu:children]
+    vgpu
+
+Having multiple groups is useful if you want to be able to do conditional
+templating in ``nova.conf`` (see :ref:`NVIDIA Kolla Ansible
+Configuration`). Since the vgpu role requires iommu to be enabled, all of the
+hosts in the ``vgpu`` group are also added to the ``iommu`` group.
+
+If using bifrost and the ``kayobe overcloud inventory discover`` mechanism,
+hosts can automatically be mapped to these groups by configuring
+``overcloud_group_hosts_map``:
+
+.. code-block:: yaml
+   :caption: ``$KAYOBE_CONFIG_PATH/overcloud.yml``
+
+    overcloud_group_hosts_map:
+      compute_vgpu:
+        - "computegpu000"
+      compute_mutli_instance_gpu:
+        - "computegpu001"
+
+.. _NVIDIA Role Configuration:
+
+Role Configuration
+^^^^^^^^^^^^^^^^^^
+
+Configure the VGPU devices:
+
+.. code-block:: yaml
+   :caption: $KAYOBE_CONFIG_PATH/inventory/group_vars/compute_vgpu/vgpu
+
+    #nvidia-692 GRID A100D-4C
+    #nvidia-693 GRID A100D-8C
+    #nvidia-694 GRID A100D-10C
+    #nvidia-695 GRID A100D-16C
+    #nvidia-696 GRID A100D-20C
+    #nvidia-697 GRID A100D-40C
+    #nvidia-698 GRID A100D-80C
+    #nvidia-699 GRID A100D-1-10C
+    #nvidia-700 GRID A100D-2-20C
+    #nvidia-701 GRID A100D-3-40C
+    #nvidia-702 GRID A100D-4-40C
+    #nvidia-703 GRID A100D-7-80C
+    #nvidia-707 GRID A100D-1-10CME
+    vgpu_definitions:
+        # Configuring a MIG backed VGPU
+        - pci_address: "0000:17:00.0"
+          virtual_functions:
+            - mdev_type: nvidia-700
+              index: 0
+            - mdev_type: nvidia-700
+              index: 1
+            - mdev_type: nvidia-700
+              index: 2
+            - mdev_type: nvidia-699
+              index: 3
+          mig_devices:
+            "1g.10gb": 1
+            "2g.20gb": 3
+        # Configuring a card in a time-sliced configuration (non-MIG backed)
+        - pci_address: "0000:65:00.0"
+          virtual_functions:
+            - mdev_type: nvidia-697
+              index: 0
+            - mdev_type: nvidia-697
+              index: 1
+
+Running the playbook
+^^^^^^^^^^^^^^^^^^^^
```

Comment:
I think there is upstream support for this now, so it's part of host configure, instead of being a separate playbook. This section can be deleted